File: 5. Longest Palindromic Substring/5.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        def funcionExpansion(izq, der, word):
            while (izq >= 0 and der < len(s) and s[izq] == s[der]):
                
                palindrome = s[izq:der+1]

                if (len(palindrome) > len(word)):
                    word = palindrome
                izq-=1
                der+=1

            return word

        word = ''
        for i in range (len(s)):
            word = funcionExpansion(i, i, word)

            word = funcionExpansion(i, i + 1, word)

        return word
    

File: 5. Longest Palindromic Substring/5.js
var longestPalindrome = function (s) {
  var word = '';

  const funcionExpansion = (izq, der) => {
      while (izq >= 0 && der < s.length && s[izq] === s[der]) {

          const palindrome = s.substring(izq, der + 1);

          if (palindrome.length > word.length) {
              word = palindrome;
          }

          izq--;
          der++;

      }
  };

  for (let i = 0; i < s.length; i++) {
      funcionExpansion(i, i); // Impares

      funcionExpansion(i, i + 1); // Pares
  }

  return word
};

module.exports = { longestPalindrome };
